### PR TITLE
Core add GetSystem functions base strategy

### PR DESCRIPTION
**Pull request by adityaghantasala**
Following the discussion in #6748 

This PR adds the `GetSystemMatrix` , `GetSystemVector` , `GetSolutionVector` methods to the base solution strategy. This is required to do 
`norm_dx = TSparseSpace::TwoNorm(mpMomentumStrategy->GetSolutionVector());`

### CODE REVIEW

File: kratos/solving_strategies/strategies/solving_strategy.h [hunk 1]
```diff
@@ -393,6 +393,46 @@ class SolvingStrategy
         KRATOS_CATCH("")
     }
 
+    /**
+     * @brief This method returns the LHS matrix
+     * @return The LHS matrix
+     */
+    TSystemMatrixType& GetSystemMatrix()
```
Comment (loumalouomega) [inline kratos/solving_strategies/strategies/solving_strategy.h]:
```suggestion
    virtual TSystemMatrixType& GetSystemMatrix()
```

File: kratos/solving_strategies/strategies/solving_strategy.h [hunk 2]
```diff
@@ -393,6 +393,46 @@ class SolvingStrategy
...
+        KRATOS_ERROR << "GetSystemMatrix not implemented in base SolvingStrategy" << std::endl;
+
+        KRATOS_CATCH("");
+    }
+
+    /**
+     * @brief This method returns the RHS vector
+     * @return The RHS vector
+     */
+    TSystemVectorType& GetSystemVector()
```
Comment (loumalouomega) [inline kratos/solving_strategies/strategies/solving_strategy.h]:
```suggestion
    virtual TSystemVectorType& GetSystemVector()
```

File: kratos/solving_strategies/strategies/solving_strategy.h [hunk 3]
```diff
@@ -393,6 +393,46 @@ class SolvingStrategy
...
+        KRATOS_ERROR << "GetSystemVector not implemented in base SolvingStrategy" << std::endl;
+
+        KRATOS_CATCH("");
+    }
+
+    /**
+     * @brief This method returns the solution vector
+     * @return The Dx vector
+     */
+    TSystemVectorType& GetSolutionVector()
```
Comment (loumalouomega) [inline kratos/solving_strategies/strategies/solving_strategy.h]:
```suggestion
    virtual TSystemVectorType& GetSolutionVector()
```

**Review by loumalouomega — CHANGES_REQUESTED** [review]
Missing virtual , and override the functions in the derived classes (expose them to python BTW)

**Comment by adityaghantasala** [comment]
Sorry forgot ! Added !

**Comment by adityaghantasala** [comment]
By the way there are some implementations of these methods in applications. What is preferred ? to make the changes here or in a different PR ?

**Review by loumalouomega — DISMISSED** [review]

**Comment by loumalouomega** [comment]
> By the way there are some implementations of these methods in applications. What is preferred ? to make the changes here or in a different PR ?

If there not many in this PR, otherwise in a independent one (if CI works)

**Comment by philbucher** [comment]
why wasn't this merged yet?

**Comment by loumalouomega** [comment]
> why wasn't this merged yet?

Was not approved, and now there is conflict

**Review by philbucher — DISMISSED** [review]

**Comment by philbucher** [comment]
sth went wrong with the merge

**Review by philbucher — APPROVED** [review]

**Comment by adityaghantasala** [comment]
@loumalouomega  thanks for merging the master !! 
@philbucher thanks for approving !

**Pull request merged**